### PR TITLE
Extensive refactor of the views, with style changes

### PR DESCRIPTION
**Pull request by alcueca**
I refactored the functions leading to `dollarBalances()`, which I assume is the one the AMO needs to provide since it's the one in the interface.

I'm not really sure if the function should show the value of the assets controlled by the AMO in USD. In that case we still need a few changes. At the very least, I understand what the previous functions did and we can discuss what do we need to show.

### CODE REVIEW

**Comment by alcueca** [comment]
Now this PR includes a valuation of the AMO assets in Frax terms, and can deal with post-maturity scenarios. When burning fyFrax the AMO will either repay debt and withdraw collateral, or redeem fyFrax for Frax, or store the fyFrax, as necessary.

**Comment by alcueca** [comment]
The AMO could probably be improved by splitting the Yield integration functions into internal and external:
The external functions read the series from the mapping and pass it on to the internal functions. This saves gas.
The external functions don't have a `to` parameter, removing a few footguns.
The external functions call the internal functions, that have the actual logic.
The internal functions only call other internal functions, saving gas.

File: src/YieldSpaceAMO.sol [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+        uint256 fyFraxAmount
+    ) public view returns (uint256 fraxAmount) {
+        Series storage _series = series[seriesId];
+        uint256 debt = cauldron.balances(series[seriesId].vaultId).art;
+        if (debt > fyFraxAmount) {
+            fraxAmount = fyFraxAmount;
+        } else {
+            fraxAmount =
+                debt +
+                _series.pool.sellFYTokenPreview((fyFraxAmount - debt).u128());
```
Comment (aniemerg) [inline src/YieldSpaceAMO.sol]: what happens if sellFYTokenPreview() is provided a value greater than it can trade?

File: src/YieldSpaceAMO.sol [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+    /// @notice mint fyFrax using FRAX as collateral 1:1 Frax to fyFrax
+    /// @dev The Frax to work with needs to be in the AMO already.
+    /// @param _series fyFrax series being minted
+    /// @param to destination for the fyFrax
+    /// @param fraxAmount amount of Frax being used to mint fyFrax at 1:1
+    function _mintFyFrax(
+        Series memory _series,
+        address to,
+        uint128 fraxAmount
+    ) public {
```
Comment (aniemerg) [inline src/YieldSpaceAMO.sol]: How can this be public?

Comment (alcueca) [reply src/YieldSpaceAMO.sol]: Fixed, well spotted

File: src/YieldSpaceAMO.sol (hunk 1 above)
Comment (alcueca) [reply src/YieldSpaceAMO.sol]: This would revert. I added a TODO, and will write about the conditions in which that would happen.

**Pull request merged**